src/selectors/profile.js

```diff
@@ -821,7 +822,7 @@ export const getThreadIdToNameMap: Selector<Map<Tid, string>> = createSelector(
   (threads) => {
     const threadIdToNameMap = new Map();
     for (const thread of threads) {
-      threadIdToNameMap.set(thread.tid, thread.name);
+      threadIdToNameMap.set(thread.tid, getFriendlyThreadName(threads, thread));
```

Comment:
The only concern I have is that `getFriendlyThreadName` does a `filter` operation on all threads, as a result this algorithm is O(n²) on the threads length. This isn't great and we might want to revisit this later (or now?).